### PR TITLE
Cache `.venv` on windows job

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/10068?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/10068/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 10068
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Package installation takes 10 ~ 15 minutes on windows:

https://github.com/mlflow/mlflow/actions/runs/6609145501/job/17948848689

image

This PR is an attempt to speed it up.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .github/actions/cache-pip/action.yml [hunk 1]
```diff
@@ -11,5 +11,5 @@ runs:
       env:
         SEGMENT_DOWNLOAD_TIMEOUT_MINS: 1
       with:
-        path: ~/.venv
+        path: .venv
```
Comment (harupy) [inline .github/actions/cache-pip/action.yml:14]: Not sure why I chose `~/.venv` when I created this action, but `.venv` seems more common.

Comment (harupy) [reply .github/actions/cache-pip/action.yml:14]: I remember why I chose `~/.venv`.

**Review by serena-ruan — APPROVED** [review]
LGTM!

File: .github/actions/cache-pip/action.yml (hunk 1 above)
Comment (serena-ruan) [reply .github/actions/cache-pip/action.yml]: Why?

Comment (harupy) [reply .github/actions/cache-pip/action.yml:14]: TLDR: MLflow adds a git-commit tag when the pytest binary is under the repo:

https://github.com/mlflow/mlflow/actions/runs/6609822686/job/17950619474

```
FAILED | MEM 4.2/6.8 GB | DISK 49.9/83.2 GB tests/store/tracking/test_rest_store.py::test_requestor - AssertionError: assert [{'key': 'mlf... 'mock user'}] == [{'key': 'mlf... 'mock user'}]
  At index 1 diff: {'key': 'mlflow.source.name', 'value': 'rest test'} != {'key': 'mlflow.source.git.commit', 'value': '0125ee338d0da139f5f37b63d62b607213a1d90a'}
  Right contains one more item: {'key': 'mlflow.user', 'value': 'mock user'}
  Full diff:
    [
     {'key': 'mlflow.runName',
      'value': 'my name'},
  -  {'key': 'mlflow.source.git.commit',
  -   'value': '0125ee338d0da139f5f37b63d62b607213a1d90a'},
     {'key': 'mlflow.source.name',
      'value': 'rest test'},
     {'key': 'mlflow.source.type',
      'value': 'LOCAL'},
     {'key': 'mlflow.user',
      'value': 'mock user'},
    ]
```

Comment (harupy) [reply .github/actions/cache-pip/action.yml:14]: @serena-ruan https://github.com/mlflow/mlflow/pull/10068/commits/f3a89671af14d19cae64f93a413b873c59962a33 should fix this.

File: mlflow/utils/git_utils.py [hunk 2]
```diff
@@ -46,6 +46,8 @@ def get_git_commit(path: str) -> Optional[str]:
         if os.path.isfile(path):
             path = os.path.dirname(path)
         repo = Repo(path, search_parent_directories=True)
+        # if path in repo.ignored(path):
+        #     return None
```
Comment (harupy) [inline mlflow/utils/git_utils.py:50]:
```suggestion
        if path in repo.ignored(path):
            return None
```

**Pull request merged**